src/linear_algebra/determinant.lean

```diff
@@ -193,6 +193,37 @@ begin
   exact ((det_row_multilinear : alternating_map R (n → R) R n).map_perm M σ).trans this,
 end
 
+/-- Permuting rows and columns with the same equivalence has no effect. -/
+@[simp]
+lemma det_minor_equiv_self (e : n ≃ m) (A : matrix m m R) :
+  det (A.minor e e) = det A :=
+begin
+  rw [det_apply', det_apply'],
+  apply finset.sum_bij' (λ σ _, equiv.perm_congr e σ) _ _ (λ σ _, equiv.perm_congr e.symm σ),
+  { intros σ _, ext, simp only [equiv.symm_symm, equiv.perm_congr_apply, equiv.symm_apply_apply] },
+  { intros σ _, ext, simp only [equiv.symm_symm, equiv.perm_congr_apply, equiv.apply_symm_apply] },
+  { intros σ _, apply finset.mem_univ },
+  { intros σ _, apply finset.mem_univ },
+  intros σ _,
+  simp_rw [equiv.perm_congr_apply],
+  rw equiv.perm.sign_perm_congr e σ,
+  congr' 1,
+  apply finset.prod_bij' (λ i _, e i) _ _ (λ i _, e.symm i),
+  { intros, simp_rw equiv.symm_apply_apply },
+  { intros, simp_rw equiv.apply_symm_apply },
+  { intros, apply finset.mem_univ },
+  { intros, apply finset.mem_univ },
+  { intros, simp_rw equiv.symm_apply_apply, rw minor_apply, },
```

Comment:
This reminded me that to define a `fintype.prod_equiv` which fills in all the arguments for `prod_bij'` over `univ` where the map is already an `equiv`: #7109. If you want to get this merged soon, I can apply `prod_equiv` in #7109 after this PR gets merged, or you can make this PR depend on #7109.

Reply:
Note that this proof is basically copy-pasted from the other file, with the only modification being the generalization of `e.symm` to `e` - so I'd prefer that you leave fixing this to #7109.